src/domain/relayer/mod.rs

```diff
@@ -436,19 +448,64 @@ impl RelayerFactoryTrait for RelayerFactory {
                     })?;
 
                 let network = MidnightNetwork::try_from(network_repo)?;
+                let indexer_urls = network.indexer_urls.clone();
+                let metadata = HashMap::from([
+                    (
+                        "network".to_string(),
+                        format!("{:?}", to_midnight_network_id(&relayer.network)),
+                    ),
+                    ("http".to_string(), indexer_urls.http),
+                    ("ws".to_string(), indexer_urls.ws),
+                ]);
+
+                let midnight_provider: MidnightProvider = get_network_provider(
+                    &network,
+                    relayer.custom_rpc_urls.clone(),
+                    Some(&metadata),
+                )?;
+
+                // Create the Midnight signer to get the wallet seed
+                let midnight_signer = MidnightSignerFactory::create_midnight_signer(&signer)
+                    .map_err(|e| {
+                        RelayerError::NetworkConfiguration(format!(
+                            "Failed to create signer: {}",
+                            e
+                        ))
+                    })?;
+                let wallet_seed = midnight_signer.wallet_seed();
+
+                // Get the indexer client from provider
+                let indexer_client = midnight_provider.get_indexer_client();
+
+                // Convert network to NetworkId for sync manager
+                let network_id = to_midnight_network_id(&relayer.network);
+
+                // Create the sync manager
+                let sync_manager = Arc::new(
+                    SyncManager::<QuickSyncStrategy>::new(indexer_client, wallet_seed, network_id)
+                        .map_err(|e| {
+                            RelayerError::NetworkConfiguration(format!(
+                                "Failed to create sync manager: {}",
+                                e
+                            ))
+                        })?,
+                );
+
+                // Signer repository is passed from the factory
 
-                let midnight_provider =
-                    get_network_provider(&network, relayer.custom_rpc_urls.clone())?;
                 let transaction_counter_service = Arc::new(TransactionCounterService::new(
                     relayer.id.clone(),
                     relayer.address.clone(),
                     transaction_counter_store,
                 ));
+
                 let dependencies = MidnightRelayerDependencies::new(
                     relayer_repository,
                     networks_repository,
                     transaction_repository,
+                    signer_repository,
```

Comment:
We can just pass midnight_signer if repository is used just to create it in midnight_relayer.